src/hypothesis/searchstrategy/lazy.py

```diff
@@ -95,30 +93,29 @@ def calc_has_reusable_values(self):
     @property
     def wrapped_strategy(self):
         if self.__wrapped_strategy is None:
-            with self.__settings:
-                unwrapped_args = tuple(
-                    unwrap_strategies(s) for s in self.__args)
-                unwrapped_kwargs = {
-                    k: unwrap_strategies(v)
-                    for k, v in self.__kwargs.items()
-                }
-
-                base = self.__function(
-                    *self.__args, **self.__kwargs
-                )
-                if (
-                    unwrapped_args == self.__args and
-                    unwrapped_kwargs == self.__kwargs
-                ):
-                    self.__wrapped_strategy = base
-                else:
-                    self.__wrapped_strategy = self.__function(
-                        *unwrapped_args,
-                        **unwrapped_kwargs)
-                    self.__wrapped_strategy.force_has_reusable_values = \
-                        base.has_reusable_values
-                    assert self.__wrapped_strategy.has_reusable_values == \
-                        base.has_reusable_values
+            unwrapped_args = tuple(
+                unwrap_strategies(s) for s in self.__args)
+            unwrapped_kwargs = {
+                k: unwrap_strategies(v)
+                for k, v in self.__kwargs.items()
+            }
+
+            base = self.__function(
+                *self.__args, **self.__kwargs
+            )
+            if (
+                unwrapped_args == self.__args and
+                unwrapped_kwargs == self.__kwargs
+            ):
+                self.__wrapped_strategy = base
+            else:
+                self.__wrapped_strategy = self.__function(
+                    *unwrapped_args,
+                    **unwrapped_kwargs)
+                self.__wrapped_strategy.force_has_reusable_values = \
+                    base.has_reusable_values
+                assert self.__wrapped_strategy.has_reusable_values == \
```

Comment:
Err… why is this assert interesting?

Reply:
Stupid reasons basically. I should improve that code, but right now if the result had previously been calculated then the force doesn't do anything.